polkadot/node/overseer/src/lib.rs

```diff
@@ -839,9 +839,6 @@ where
 			self.on_head_deactivated(deactivated)
 		}
 
-		self.broadcast_signal(OverseerSignal::BlockFinalized(block.hash, block.number))
```

Comment:
I think we should also skip the above active leaves pruning.

Reply:
Would it be fair? I mean, don't we leak more memory this way than would be leaked due to backoff disabling only?